superduperdb/components/datatype.py

```diff
@@ -185,24 +196,36 @@ def encode(
 
         def _encode(x):
             try:
-                bytes_ = self.datatype.encoder(x)
+                x = self.datatype.encoder(x)
             except Exception as e:
-                raise ArtifactSavingError from e
-            sha1 = str(hashlib.sha1(bytes_).hexdigest())
-            if (
-                CFG.bytes_encoding == BytesEncoding.BASE64
-                or bytes_encoding == BytesEncoding.BASE64
-            ):
-                bytes_ = to_base64(bytes_)
-            return bytes_, sha1
+                raise ArtifactSavingError(e) from e
+
+            if isinstance(x, str):
+                sha1 = str(hashlib.sha1(x.encode()).hexdigest())
+            elif isinstance(x, bytes):
+                sha1 = str(hashlib.sha1(x).hexdigest())
+                if (
+                    CFG.bytes_encoding == BytesEncoding.BASE64
+                    or bytes_encoding == BytesEncoding.BASE64
+                ):
+                    # TODO: artifact stores is not compatible with non-bytes types
+                    x = to_base64(x)
```

Comment:
If we use BASE64, the x will be a string, and we can not  save it with using `_save_bytes`

Reply:
Ok, some databases don't support `bytes`, so we need this alternative.

Reply:
Maybe we should distinguish between `artifact=True/False`. The idea is that we only use `BASE64` when `artifact=False`